### PR TITLE
PRL-4535-cleanup-solicitor-data

### PR DESCRIPTION
**Pull request by sahaSonali**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRL-4535

### Change description ###
Before submitting the below events, data clean up will happeing based on the Yes/No/Dont know selection

    C100 case creation - Applicant details/ Respondent details/Child details
    C100 case amendment - Amend either of Applicant details/ Respondent details
    FL401 case creation - Applicant details and Respondent details
    FL401 case amendment - Amend either of Applicant details and Respondent details

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/services/UpdatePartyDetailsService.java [hunk 1]
```diff
@@ -107,10 +117,91 @@ public Map<String, Object> updateApplicantRespondentAndChildData(CallbackRequest
                 setApplicantOrganisationPolicyIfOrgEmpty(updatedCaseData, ElementUtils.unwrapElements(applicantList.get()).get(0));
             }
         }
-
+        cleanUpCaseDataBasedOnYesNoSelection(updatedCaseData, caseData);
+        log.info("CaseData Respondents " + caseData.getRespondents());
+        log.info("CaseData RespondentsFL401 " + caseData.getRespondentsFL401());
+        log.info("Map Respondents " + updatedCaseData.get("respondents"));
+        log.info("Map RespondentsFL401 " + updatedCaseData.get("respondentsFL401"));
         return updatedCaseData;
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/services/UpdatePartyDetailsService.java]: please remove all the logger statements.

**Review by AlokDatta — CHANGES_REQUESTED** [review]
Please update the PR details as well. thank you

File: src/main/java/uk/gov/hmcts/reform/prl/services/UpdatePartyDetailsService.java (hunk 1 above)
Comment (sahaSonali) [reply src/main/java/uk/gov/hmcts/reform/prl/services/UpdatePartyDetailsService.java]: Removed

**Review by AlokDatta — APPROVED** [review]

**Review by shashisk97 — APPROVED** [review]
LGTM

**Pull request merged**